packages/core/src/page.js

```diff
@@ -1,124 +1,40 @@
+    session.send('Runtime.enable').catch(session._handleClosedError);
 
-    // if there is a parent session, automatically init this session
-    this.parent = browser.pages.get(parentId);
-
-    if (this.parent) {
-      this.parent.frames.add(this);
-      this._handleCloseRace(this.init(this.parent.options));
-    }
+    this.log.debug('Page created');
   }
 
-  // initial page options asynchronously
-  async init(options = {}) {
-    this.options = options;
-
-    let {
-      cacheDisabled = true,
-      enableJavaScript = true,
-      requestHeaders = {},
-      networkIdleTimeout,
-      authorization,
-      userAgent,
-      intercept,
-      meta
-    } = options;
-
-    this.log.debug('Initialize page', meta);
-    this.network.timeout = networkIdleTimeout;
-    this.network.authorization = authorization;
-    this.meta = meta;
-
-    let [, { frameTree }, version] = await Promise.all([
-      this.send('Page.enable'),
-      this.send('Page.getFrameTree'),
-      this.send('Browser.getVersion')
-    ]);
-
-    this.frameId = frameTree.frame.id;
-    // by default, emulate a non-headless browser
-    userAgent ||= version.userAgent.replace('Headless', '');
-
-    // auto-attach related targets
-    let autoAttachTarget = {
-      waitForDebuggerOnStart: false,
-      autoAttach: true,
-      flatten: true
-    };
-
-    await Promise.all([
-      this.send('Runtime.enable'),
-      this.send('Target.setAutoAttach', autoAttachTarget),
-      this.send('Page.setLifecycleEventsEnabled', { enabled: true }),
-      this.send('Network.setCacheDisabled', { cacheDisabled }),
-      this.send('Network.setExtraHTTPHeaders', { headers: requestHeaders }),
-      this.send('Network.setUserAgentOverride', { userAgent }),
-      this.send('Security.setIgnoreCertificateErrors', { ignore: true }),
-      this.send('Emulation.setScriptExecutionDisabled', { value: !enableJavaScript })
-    ]);
-
-    if (intercept) {
-      await this.network.intercept(intercept);
-    }
-
-    return this;
-  }
-
-  // Close the target page if not already closed
+  // Close the page
   async close() {
-    if (!this.browser) return;
-
-    this.log.debug('Page closing', this.meta);
-
-    /* istanbul ignore next: errors race here when the browser closes */
-    await this.browser.send('Target.closeTarget', { targetId: this.targetId })
-      .catch(error => this.log.debug(error, this.meta));
+    await this.session.close();
+    this.log.debug('Page closed', this.meta);
   }
 
-  async resize({
-    deviceScaleFactor = 1,
-    mobile = false,
-    height,
-    width
-  }) {
+  // Resize the page to the specified width and height
+  async resize({ width, height }) {
     this.log.debug(`Resize page to ${width}x${height}`);
 
-    await this.send('Emulation.setDeviceMetricsOverride', {
-      deviceScaleFactor,
-      mobile,
+    await this.session.send('Emulation.setDeviceMetricsOverride', {
+      deviceScaleFactor: 1,
+      mobile: false,
```

Comment:
One day, we'll probably need to make this configurable (probably when we capture many DOM snapshots per-snapshot). Until then...! 🏃🏼‍♂️